ThreeDSecure/src/test/java/com/braintreepayments/api/ThreeDSecureClientUnitTest.java

```diff
@@ -317,50 +348,285 @@ public void onBrowserSwitchResult_whenFailure_postsErrorWithResponse() throws Ex
-        BraintreeClient braintreeClient = new MockBraintreeClientBuilder().build();
-
-        ThreeDSecureClient sut = new ThreeDSecureClient(braintreeClient, cardinalClient, browserSwitchHelper);
-        sut.onBrowserSwitchResult(null, threeDSecureResultCallback);
+        BraintreeClient braintreeClient = new MockBraintreeClientBuilder()
+                .deliverBrowserSwitchResult(null)
+                .build();
 
-        ArgumentCaptor<Exception> captor = ArgumentCaptor.forClass(Exception.class);
-        verify(threeDSecureResultCallback).onResult((ThreeDSecureResult) isNull(), captor.capture());
+        ThreeDSecureClient sut = new ThreeDSecureClient(activity, lifecycle, braintreeClient, cardinalClient, browserSwitchHelper, threeDSecureAPI);
+        sut.setListener(listener);
+        sut.onBrowserSwitchResult(activity);
 
-        Exception exception = captor.getValue();
-        assertTrue(exception instanceof BraintreeException);
-        assertEquals("BrowserSwitchResult cannot be null", exception.getMessage());
+        verify(listener, never()).onThreeDSecureFailure(any(Exception.class));
+        verify(listener, never()).onThreeDSecureSuccess(any(ThreeDSecureResult.class));
     }
 
     @Test
     public void onBrowserSwitchResult_whenBrowserSwitchStatusCanceled_returnsExceptionToCallback() {
-        CardinalClient cardinalClient = new MockCardinalClientBuilder().build();
-        BraintreeClient braintreeClient = new MockBraintreeClientBuilder().build();
-
         BrowserSwitchResult browserSwitchResult =
                 new BrowserSwitchResult(BrowserSwitchStatus.CANCELED, null, null);
 
-        ThreeDSecureClient sut = new ThreeDSecureClient(braintreeClient, cardinalClient, browserSwitchHelper);
-        sut.onBrowserSwitchResult(browserSwitchResult, threeDSecureResultCallback);
+        CardinalClient cardinalClient = new MockCardinalClientBuilder().build();
+        BraintreeClient braintreeClient = new MockBraintreeClientBuilder()
+                .deliverBrowserSwitchResult(browserSwitchResult)
+                .build();
+
+        ThreeDSecureClient sut = new ThreeDSecureClient(activity, lifecycle, braintreeClient, cardinalClient, browserSwitchHelper, threeDSecureAPI);
+        sut.setListener(listener);
+        sut.onBrowserSwitchResult(activity);
 
         ArgumentCaptor<Exception> captor = ArgumentCaptor.forClass(Exception.class);
-        verify(threeDSecureResultCallback).onResult((ThreeDSecureResult) isNull(), captor.capture());
+        verify(listener).onThreeDSecureFailure(captor.capture());
+        verify(listener, never()).onThreeDSecureSuccess(any(ThreeDSecureResult.class));
 
         Exception exception = captor.getValue();
         assertTrue(exception instanceof UserCanceledException);
         assertEquals("User canceled 3DS.", exception.getMessage());
     }
+
+    @Test
+    public void onBrowserSwitchResult_whenListenerNull_doesNothing() {
```

Comment:
Marking a note here to call out our questions noted in this test.

Reply:
We could no-op when the listener is `null` and then defer delivery until a listener is attached.

Reply:
Ah that's a good point. For now i'd say let's assume the listener is set. It's a two-liner integration pattern there's bound to be some weirdness somewhere lol. We can re-evaluate but I'd bank on that listener being set to keep the scope of this PR narrow and maybe place a TODO here and circle back before GA?